modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/util/BeanUtil.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.util;
+
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Service;
+
+/**
+ * <p>
+ * We cannot inject any Spring-managed bean in any of the provider classes because these are instantiated by JPA before Spring injects anything into it.
+ * This is a utility class to fetch any bean according to our requirements.
+ * </p>
+ */
+@Service
+public class BeanUtil implements ApplicationContextAware {
```

Comment:
Not excited about this class but I understand you're doing it as a hack. I'll take a closer look to see where you are using this and to see whether there are alternatives.

This should be a `@Component` rather than a `@Service`, though. A service is more for domain logic, and that's not what this is doing.

Reply:
Thanks for calling that out. This is fixed.